linkerd/tls/src/server.rs

```diff
@@ -107,27 +84,27 @@ impl<L, P, N> NewDetectTls<L, P, N> {
     }
 }
 
-impl<T, L, P, N> NewService<T> for NewDetectTls<L, P, N>
+impl<T, L, P, N> NewService<(T, Option<ServerName>)> for NewDetectTls<L, P, N>
 where
-    P: ExtractParam<Timeout, T> + ExtractParam<L, T> + Clone,
+    P: ExtractParam<L, T> + Clone,
     N: Clone,
 {
     type Service = DetectTls<T, L, P, N>;
 
-    fn new_service(&self, target: T) -> Self::Service {
-        let timeout = self.params.extract_param(&target);
+    fn new_service(&self, t: (T, Option<ServerName>)) -> Self::Service {
+        let (target, sni) = t;
         let local_identity = self.params.extract_param(&target);
         DetectTls {
             target,
             local_identity,
-            timeout,
+            sni,
             params: self.params.clone(),
             inner: self.inner.clone(),
         }
     }
 }
 
-impl<I, T, L, LIo, P, N, NSvc> Service<I> for DetectTls<T, L, P, N>
+impl<I, T, L, LIo, P, N, NSvc> Service<DetectIo<I>> for DetectTls<T, L, P, N>
```

Comment:
This no longer _detects_ really, right? Should this be renamed to reflect what it does?

It's also pretty awkward to me that we pass an optional SNI and the local identity here and then at process-time we check whether we should actually do TLS termination.